packages/migration/migrations/hearth/20230602120944-populate-valid-locationid.js

```diff
@@ -0,0 +1,173 @@
+
+    while (processedDocCount < totalCompositionsWithoutLocationIds) {
+      const elasticDocBatchResult = await searchCompositionByCriteria(
+        searchCriteria,
+        { size: 10 }
+      )
+      const elasticDocBatch = elasticDocBatchResult.body.hits.hits
+      // eslint-disable-next-line no-console
+      console.log(
+        `Migration - ElasticSearch :: Processing ${processedDocCount + 1} - ${
+          processedDocCount + limit
+        } of ${totalCompositionsWithoutLocationIds} compositions...`
+      )
+
+      for (const elasticDoc of elasticDocBatch) {
+        // populate the missing eventCountry and eventLocationId/eventJurisdictionIds for each doc
+        await updateEventLocationIdOrJurisdictionIds(db, elasticDoc)
+      }
+      processedDocCount += limit
+    }
+    // eslint-disable-next-line no-console
+    console.log(
+      `Migration - ElasticSearch :: Process for populating missing eventLocationId/eventJurisdictionIds  completed successfully.`
+    )
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `Migration - ElasticSearch :: Process for populating missing eventLocationId/eventJurisdictionIds failed : ${error.stack}`
+    )
+  } finally {
+    await session.endSession()
+  }
+}
+
+export const down = async (db, client) => {
+  // TODO write the statements to rollback your migration (if possible)
+  // Example:
+  // await db.collection('albums').updateOne({artist: 'The Beatles'}, {$set: {blacklisted: false}});
+}
+
+async function updateEventLocationIdOrJurisdictionIds(db, elasticDoc) {
+  const body = {}
+  const compositionId = elasticDoc._id
+  const compositionFromDB = await getCollectionDocuments(
+    db,
+    COLLECTION_NAMES.COMPOSITION,
+    [compositionId]
+  )
+
+  if (!compositionFromDB) {
+    throw new Error(`could not find composition with id: ${compositionId}`)
+  }
+
+  const composition = compositionFromDB[0]
+  const encounterSection = composition.section.find(
+    (section) =>
+      section.code.coding[0].code === 'birth-encounter' ||
+      section.code.coding[0].code === 'death-encounter'
+  )
+
+  if (!encounterSection) {
+    throw new Error(
+      `could not find encounter in compositionID: ${compositionId}`
+    )
+  }
+  const encounterId = encounterSection.entry[0]?.reference.replace(
+    'Encounter/',
+    ''
+  )
+
+  const encounterDoc = await getCollectionDocuments(
+    db,
+    COLLECTION_NAMES.ENCOUNTER,
+    [encounterId]
+  )
+
+  if (encounterDoc.length > 0) {
+    const locationId = encounterDoc[0].location[0].location.reference.replace(
+      'Location/',
+      ''
+    )
+
+    const locationDoc = await getCollectionDocuments(
+      db,
+      COLLECTION_NAMES.LOCATION,
+      [locationId]
+    )
+
+    if (locationDoc.length > 0) {
+      if (locationDoc[0].type.coding[0].code === 'HEALTH_FACILITY') {
+        body.eventLocationId = locationDoc[0].id
+        body.eventCountry = 'FAR'
```

Comment:
We probably don't need the eventCountry for 'HEALTH_FACILITY'. As we already know it's in our location hierarchy. Could you still double check in search microservice please? That way we can avoid quite a bit of hassle